### ISSUE TITLE
[feature request] Add CNCF Milestones

### ISSUE THREAD

**Issue by mfdii** [cncf/devstats.archive#209]
It would be nice if we could overlay CNCF milestones (sandbox, incubation, graduation), much like we can overlay releases, onto the various graphs. This makes it easier to see the growth of a project since they hit a particular milestone.

**Comment by lukaszgryglicki** [comment]
I will add this, but first of all I need those milestone dates for all CNCF projects (like for example Kubernetes: I already have join date, now I would need Sandbox join date, then Incubation date and finally graduation date, same for all other projects - not all projects would have Sandbox date, some started as Incubating).
cc @dankohn

**Comment by lukaszgryglicki** [comment]
I will work on that, but I'm treating this as a low priority task and I also need a list of dates of joining Sandbox (if present), promoting to Incubation or joining as incubating (if present), promoting to Graduated (if present) for all CNCF projects.
I'm blocked on having all those dates first.

**Comment by lukaszgryglicki** [comment]
I've got data I need, starting to work on this.

**Comment by lukaszgryglicki** [comment]
Updated code to handle new milestones, like moving to incubation, graduating, archiving. In charts/graphs dashboards you will have new annotation lines, for example: 
![Screenshot 2019-09-24 at 10 58 32](https://user-images.githubusercontent.com/2469783/65497439-8e0cd600-deba-11e9-949c-0646622b0465.png)
![Screenshot 2019-09-24 at 10 59 17](https://user-images.githubusercontent.com/2469783/65497441-8e0cd600-deba-11e9-99a3-baac9c834a5d.png)
![Screenshot 2019-09-24 at 11 02 27](https://user-images.githubusercontent.com/2469783/65497633-d4facb80-deba-11e9-9a6f-8ff7c5b4515e.png)

In tabular dashboards there will be new ranges, like this:
![Screenshot 2019-09-24 at 10 59 46](https://user-images.githubusercontent.com/2469783/65497498-a3820000-deba-11e9-8925-98b089b97327.png)

Possible new ranges are:

- CNCF join - moving to incubation
- CNCF join - graduating (for projects that started in incubating state)
- Moving to incubation - graduating (only CoreDNS)
- Since moving to incubation
- Since graduating.

I'm not adding ranges related to archiving, because there is no clear info of when archiving can happen during this process (I mean what is the correct order for archiving - it can happen anytime). I'm only adding `Archived` annotation line in charts/graphs (just a vertical line).

*Please also note that there is no data for new ranges in tabular views yet*.

**Comment by lukaszgryglicki** [comment]
Data also generated, few examples:
- https://k8s.devstats.cncf.io/d/24/overall-project-statistics?orgId=1&var-period_name=CNCF%20join%20date%20-%20graduated&var-repogroup_name=All
- https://k8s.devstats.cncf.io/d/24/overall-project-statistics?orgId=1&var-period_name=Since%20graduating&var-repogroup_name=All
- https://rook.devstats.cncf.io/d/9/developer-activity-counts-by-repository-group-table?orgId=1&var-period_name=CNCF%20join%20date%20-%20moved%20to%20incubation&var-metric=contributions&var-repogroup_name=All&var-country_name=All
- https://rook.devstats.cncf.io/d/9/developer-activity-counts-by-repository-group-table?orgId=1&var-period_name=Since%20moving%20to%20incubating%20state&var-metric=contributions&var-repogroup_name=All&var-country_name=All
- https://coredns.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=CNCF%20join%20date%20-%20moved%20to%20incubation&var-metric=contributions
- https://coredns.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Moved%20to%20incubation%20-%20graduated&var-metric=contributions
- https://coredns.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Since%20graduating&var-metric=contributions

cc @mfdii

**Comment by lukaszgryglicki** [comment]
I'm closing this feature request, please let me know if you want to reopen.
